go/posix-router/main.go

```diff
@@ -19,27 +19,100 @@ import (
 	"encoding/json"
 	"errors"
 	"fmt"
+	"net"
 	"net/http"
 	_ "net/http/pprof"
+	"time"
 
 	"github.com/go-chi/chi/v5"
 	"github.com/go-chi/cors"
 	"golang.org/x/sync/errgroup"
+	"google.golang.org/grpc"
 
 	"github.com/scionproto/scion/go/lib/log"
 	"github.com/scionproto/scion/go/lib/serrors"
 	"github.com/scionproto/scion/go/lib/topology"
 	"github.com/scionproto/scion/go/pkg/app"
 	"github.com/scionproto/scion/go/pkg/app/launcher"
+	colpb "github.com/scionproto/scion/go/pkg/proto/colibri"
 	"github.com/scionproto/scion/go/pkg/router"
 	"github.com/scionproto/scion/go/pkg/router/api"
 	"github.com/scionproto/scion/go/pkg/router/config"
 	"github.com/scionproto/scion/go/pkg/router/control"
 	"github.com/scionproto/scion/go/pkg/service"
 )
 
+const grpcServerAddr = "localhost:5045"
+const rateLimiterAsAdress = "1-ff00:0:111"
+
 var globalCfg config.Config
 
+type rateLimiterServer struct {
+	colpb.UnimplementedRateLimiterServiceServer
+	dp *router.DataPlane
+}
+
+func (r *rateLimiterServer) AddRateLimit(ctx context.Context, req *colpb.AddRateLimitRequest) (*colpb.Success, error) {
```

Comment:
Why do you need this function?
It would be more intuitive to just have the SetBurstSizeAndRate() function.
If there is no rate limit for the identifier yet, SetBurstSizeAndRate() could just add one.

Reply:
Sure